shoop/themes/classic_gray/templates/classic_gray/shoop/front/basket/partials/_basket_content.jinja

```diff
@@ -61,13 +64,16 @@
                     </h4>
                 </div>
                 {% endif %}
-                <div class="delete">
-                    {% if line.can_delete %}
-                    <button type="submit" class="btn btn-sm" name="delete_{{ line.line_id }}" title="{% trans %}Remove product from basket{% endtrans %}">
-                        <i class="fa fa-times"></i>
-                    </button>
-                    {% endif %}
-                </div>
+
+                {% if not line.line_id.startswith("free_") %}
+                    <div class="delete">
+                        {% if line.can_delete %}
```

Comment:
Put your "free_" checks in line.can_delete. Not sure if this free_ thing is ok either.